src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.MultiSegment.cs

```diff
@@ -2384,21 +2384,37 @@ private bool ConsumeCommentMultiSegment()
                 ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.InvalidCharacterAtStartOfComment, marker);
             }
 
+            bool allowEmptyComment = marker == JsonConstants.Slash;
+
             localBuffer = localBuffer.Slice(1);
             if (localBuffer.Length == 0)
             {
-                if (IsLastSpan)
-                {
-                    ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.UnexpectedEndOfDataWhileReadingComment);
-                }
-                if (!GetNextSpan())
+                if (IsLastSpan || !GetNextSpan())
                 {
                     if (IsLastSpan)
                     {
-                        ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.UnexpectedEndOfDataWhileReadingComment);
+                        if (allowEmptyComment)
+                        {
+                            if (_tokenType != JsonTokenType.Comment)
```

Comment:
The nesting is getting quite deep here. We should invert the condition and break this into a method.

Reply:
How about:
```C#
if (localBuffer.Length == 0)
{
    if (IsLastSpan)
    {
        ConsumeEmptySingleLineComment(marker);
        return true;
    }

    if (!GetNextSpan())
    {
        if (IsLastSpan)
        {
            ConsumeEmptySingleLineComment(marker);
            return true;
        }
        return false;
    }

    _totalConsumed++;
    _bytePositionInLine++;
    localBuffer = _buffer;
    leftOver = 0;
}
```

```C#
private void ConsumeEmptySingleLineComment(byte marker)
{
    if (marker != JsonConstants.Slash)
    {
        ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.UnexpectedEndOfDataWhileReadingComment);
    }

    if (_tokenType != JsonTokenType.Comment)
    {
        _previousTokenType = _tokenType;
    }
    _tokenType = JsonTokenType.Comment;
    _totalConsumed++;
    _bytePositionInLine++;
    _consumed += 2;
}
```

Reply:
initially changed to that but later have removed ConsumeEmptySingleLineComment completely - once I unified consume and skip it was almost no op (only if-statement was there)